Example-LPARActivate-C/cpp/hwirstc1.cpp

```diff
@@ -0,0 +1,1163 @@
+  endTimer = timeDay.tv_sec;
+
+  printf("elapsed time for %s completion is %.2f seconds\n",
+         description,
+         difftime(endTimer, startTimer));
+}
+
+/*
+ * Method: endTimeRecorded
+ *
+ * Print the current time.
+ */
+char *printTime()
+{
+  gettimeofday(&timeDay, NULL);
+  tvSeconds = timeDay.tv_sec;
+  localTimeREST = localtime(&tvSeconds);
+
+  return asctime(localTimeREST);
+}
+
+/*
+ * Method: isJobRunning
+ *
+ * Retrieve the status of the job associated with the passed in
+ * job uri. Return true if the job is still running, has the status
+ * of running or cancel-pending. Otherwise return false.
+ *
+ * input arguments: uri, target name
+ * output arguments: pointer to jobStatus string
+ */
+bool isJobRunning(char *uriArg, char *targetNameArg, char **jobStatus)
+{
+  REQUEST_PARM_TYPE request;
+  RESPONSE_PARM_TYPE response;
+
+  int responseBodyLen = defaultLen15MB;
+  char *responseDate = (char *)malloc(defaultLen);
+  char *requestId = (char *)malloc(defaultLen);
+  char *responseBody = (char *)malloc(responseBodyLen);
+
+  bool jobRunning = false;
+  char buffer[64];
+
+  memset(&request, 0, sizeof(REQUEST_PARM_TYPE));
+  memset(&response, 0, sizeof(RESPONSE_PARM_TYPE));
+
+  request.httpMethod = HWI_REST_GET;
+  request.uri = uriArg;
+  request.uriLen = strlen(uriArg);
+  request.targetName = targetNameArg;
+  request.targetNameLen = strlen(targetNameArg);
+  request.requestTimeout = 0x00002688;
+
+  memset(responseBody, 0, responseBodyLen);
+  memset(responseDate, 0, defaultLen);
+  memset(requestId, 0, defaultLen);
+
+  response.responseBody = responseBody;
+  response.responseBodyLen = responseBodyLen;
+  response.responseDate = responseDate;
+  response.responseDateLen = defaultLen;
+  response.requestId = requestId;
+  response.requestIdLen = defaultLen;
+
+  if (verbose2)
+  {
+    traceRequest(&request, &response);
+  }
+
+  hwirest(
+      &request,
+      &response);
+
+  if (response.httpStatus == 200 &&
+      response.responseBodyLen > 0)
+  {
+    /* Parse the response JSON text. */
+    if (parse_json_text((char *)response.responseBody))
+    {
+      *jobStatus = find_string(0, "status");
+
+      if (*jobStatus == NULL)
+      {
+        printf("Error encountered retrieving status property\n");
+      }
+      else
+      {
+        if (verbose2)
+        {
+          printf("job status is %s\n", **jobStatus);
+        }
+
+        if (strncmp(statusJobRunning, *jobStatus,
+                    strlen(statusJobRunning)) == 0 ||
+            strncmp(statusJobCanPen, *jobStatus,
+                    strlen(statusJobCanPen)) == 0)
+        {
+          jobRunning = true;
+        }
+      }
+    }
+  }
+
+  free(responseBody);
+  free(responseDate);
+  free(requestId);
+
+  return jobRunning;
+}
+
+/*
+ * Method: printConstTextStr
+ *
+ * Print out the contents of the constant string if the string
+ * length is greater than 0
+ */
+void printConstTextStr(int len, const char *text, char *description)
```

Comment:
Calls to this function could be safely replaced with `printf("* >%s: %.*s", description, len, text);`